app/components/out_of_stock_modal_component/out_of_stock_modal_component.html.haml

```diff
@@ -0,0 +1,26 @@
+%div{ id: @id, "data-controller": "modal out-of-stock-modal", "data-action": "keyup@document->modal#closeIfEscapeKey modal:closing->out-of-stock-modal#redirect", "data-modal-instant-value": @instant, "data-out-of-stock-modal-redirect-value": @redirect }
```

Comment:
Where there's more than one `data-` attribute, it's usually worth nesting them like so:
```suggestion
%div{ id: @id, data: {controller: "modal out-of-stock-modal", action: "keyup@document->modal#closeIfEscapeKey modal:closing->out-of-stock-modal#redirect", "modal-instant-value": @instant, "out-of-stock-modal-redirect-value": @redirect } }
```

Although I have to admit I sometimes get sucked into spending a lot of time rewriting attributes in haml.. so I'll stop here!